### ISSUE TITLE
About MRS+Flow+RGB testing

### ISSUE THREAD

**Issue by errllxj** [craston/MARS#12]
When I tested MRS+Flow+RGB, my accuracy was only 94.8%.May I know how you tested it? I wish you could tell me

**Comment by craston** [comment]
Could you please let me know for which dataset you achieved the above accuracy?  Here's a [link ](https://github.com/craston/MARS/issues/11#issuecomment-564530894)to the code.

**Comment by errllxj** [comment]
Thank you a lot. By the way ,could you tell me  how many GPUs you used and how long it took to train the Kinetics400 dataset?

**Comment by errllxj** [comment]
Thank you a lot. By the way ,could you tell me  how many GPUs you used and how long it took to train on the mini-Kinetics dataset and Kinetics400?

**Comment by craston** [comment]
I used 4 Nvidia V100 GPU (16GB memory each)
Training for around 250 epochs
MiniKinetics (16 frames clips) ~2-3days
MiniKinetics (64 frames clips) ~4-5days
Kinetics (16frames clips) ~9-10 days
Kinetics (64frames clips) ~30days

**Comment by errllxj** [comment]
thank you for your reply.
I want to repeat your experiment on MiniKinetics, but many videos on YouTube about this dataset have been lost.
it depressed me.I would appreciate it if you could share the MiniKinetics dataset with me.
Of course, if this is not convenient for you, I would also appreciate your previous reply.
looking forward to your reply.






------------------ 原始邮件 ------------------
发件人:&nbsp;"craston"<notifications@github.com&gt;;
发送时间:&nbsp;2020年2月27日(星期四) 下午5:37
收件人:&nbsp;"craston/MARS"<MARS@noreply.github.com&gt;;
抄送:&nbsp;"刘祥健"<1731949128@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [craston/MARS] About MRS+Flow+RGB testing (#12)




I used 4 Nvidia V100 GPU (16GB memory each)
 Training for around 250 epochs
 MiniKinetics (16 frames clips) ~2-3days
 MiniKinetics (64 frames clips) ~4-5days
 Kinetics (16frames clips) ~9-10 days
 Kinetics (64frames clips) ~30days

**Comment by craston** [comment]
Hi, 
 
I switched jobs and therefore, I no longer have access to the Mini-Kinetics dataset. Sorry, I cannot help you with this.

**Comment by errllxj** [comment]
Hi,
I'm confused now.
When I tested HMDB51 split1, split2, and split3 with "MARS_HMDB51_64f.pth", the accuracy reached 80.3%, 94.2%, and 93.2%, respectively. I observed that the data volume of the three splits is 1530 videos. I feel like something went wrong.
These test results are obviously wrong, but I tested them according to your method
I appreciate your previous reply. I hope you can help me and look forward to your letter.





------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"craston"<notifications@github.com&gt;;
发送时间:&nbsp;2020年2月28日(星期五) 晚上10:33
收件人:&nbsp;"craston/MARS"<MARS@noreply.github.com&gt;;
抄送:&nbsp;"刘祥健"<1731949128@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [craston/MARS] About MRS+Flow+RGB testing (#12)




Hi,
 
I switched jobs and therefore, I no longer have access to the Mini-Kinetics dataset. Sorry, I cannot help you with this.

**Comment by craston** [comment]
Hi,
MARS_HMDB51_64f.pth is a model that was trained on HMDB split 1. Therefore its performance on split1 is as expected.
However if you use this model to test on split-2 or 3, the model has already seen some of the split-2 or 3 test samples as these were present in split-1 training samples. As a result, you get a much higher accuracy. I have uploaded the models trained on split-2 and split-3 [here](https://drive.google.com/drive/folders/1OVhBnZ_FmqMSj6gw9yyrxJJR8yRINb_G) (HMDB51/MARS_HDMB51_2_64.pth) HMDB51/MARS_HMDB51_3_64f.pth)

**Comment by errllxj** [comment]
Hi,
Thank you for your help last time.
Because I want to repeat your experimental process, I would&nbsp;appreciate it&nbsp; if&nbsp;you&nbsp; provide RGB and optical flow models&nbsp; on HMDB51 split2 -3 and UCF101 split2 -3. I hope I didn't bother you and look forward to hearing from you




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"craston"<notifications@github.com&gt;;
发送时间:&nbsp;2020年3月5日(星期四) 下午4:05
收件人:&nbsp;"craston/MARS"<MARS@noreply.github.com&gt;;
抄送:&nbsp;"刘祥健"<1731949128@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [craston/MARS] About MRS+Flow+RGB testing (#12)




Hi,
 MARS_HMDB51_64f.pth is a model that was trained on HMDB split 1. Therefore its performance on split1 is as expected.
 However if you use this model to test on split-2 or 3, the model has already seen some of the split-2 or 3 test samples as these were present in split-1 training samples. As a result, you get a much higher accuracy. I have uploaded the models trained on split-2 anmd split-3 here (https://drive.google.com/drive/folders/1OVhBnZ_FmqMSj6gw9yyrxJJR8yRINb_G) (HMDB51/MARS_HDMB51_2_64.pth HMDB51/MARS_HMDB51_3_64f.pth)

**Comment by errllxj** [comment]
Hi,
I have solved the last problem. I retrained the network .&nbsp;
When researching your thesis, I found the video-based class activation maps interesting.&nbsp; The code of class activation maps is for&nbsp; pictures on github and your code is for video. Would you like to send me a copy of this code?
&nbsp;I am looking forward to your reply.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"craston"<notifications@github.com&gt;;
发送时间:&nbsp;2020年3月5日(星期四) 下午4:05
收件人:&nbsp;"craston/MARS"<MARS@noreply.github.com&gt;;
抄送:&nbsp;"刘祥健"<1731949128@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [craston/MARS] About MRS+Flow+RGB testing (#12)




Hi,
 MARS_HMDB51_64f.pth is a model that was trained on HMDB split 1. Therefore its performance on split1 is as expected.
 However if you use this model to test on split-2 or 3, the model has already seen some of the split-2 or 3 test samples as these were present in split-1 training samples. As a result, you get a much higher accuracy. I have uploaded the models trained on split-2 anmd split-3 here (https://drive.google.com/drive/folders/1OVhBnZ_FmqMSj6gw9yyrxJJR8yRINb_G) (HMDB51/MARS_HDMB51_2_64.pth HMDB51/MARS_HMDB51_3_64f.pth)

**Comment by craston** [comment]
Unfortunately since I moved companies, I no longer have access to the code. You can however refer to this [code](https://github.com/zhoubolei/CAM/blob/master/pytorch_CAM.py) and adapt it to view the class activation maps. I hope this helps.

**Comment by errllxj** [comment]
Hi，
When I tried to test the RGB_Something_Something_64f.pth, the accuracy was very low, only 3%. I used the data processing method you recommended, and the test results confuse me. Looking forward to your reply.

**Comment by errllxj** [comment]
Hi，
When I tried to test the RGB_Something_Something_64f.pth, the accuracy was very low, only 3%. I used the data processing method you recommended, and the test results confuse me. Looking forward to your reply.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "craston/MARS"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年5月7日(星期四) 下午3:34
收件人:&nbsp;"craston/MARS"<MARS@noreply.github.com&gt;;
抄送:&nbsp;"刘祥健"<1731949128@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [craston/MARS] About MRS+Flow+RGB testing (#12)





 
Closed #12.
